### PR TITLE
[15.0][4020][ADD] partner_edit_restrict

### PR DESCRIPTION
**Pull request by AungKoKoLin1997**
[4020](https://www.quartile.co/web#id=4020&cids=3&model=project.task&view_type=form)

### CODE REVIEW

**Comment by yostashiro** [comment]
@AungKoKoLin1997 Can you follow up on the test error please.

**Review by kanda999 — CHANGES_REQUESTED** [review]
Get an error when normal user try to create a purchase order as a vendor with a partner whose changes are restricted (`restrict_edit=True`). Can you please confirm this?

2023-11-30_12h43_28

**Review by kanda999 — APPROVED** [review]
Functional test

File: partner_edit_restrict/models/res_partner.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    )
+
+    def write(self, values):
+        if self.env.user.has_group("base.group_system"):
+            return super(ResPartner, self).write(values)
+        if self.env.is_superuser():
+            return super(ResPartner, self).write(values)
+        for record in self:
+            if record.restrict_edit:
+                raise UserError(_("You are not allowed to modify this parnter record."))
```
Comment (kanda999) [inline partner_edit_restrict/models/res_partner.py]:
```suggestion
                raise UserError(_("You are not allowed to modify this partner record."))
```

**Review by kanda999 — CHANGES_REQUESTED** [review]
I found typo

**Review by kanda999 — APPROVED** [review]

**Pull request merged**